src/main/java/javaserver/ResponseBuilders/DirectoryResponseBuilder.java

```diff
@@ -0,0 +1,48 @@
+package javaserver.ResponseBuilders;
+
+import java.util.Arrays;
+
+import javaserver.HTMLContent;
+import javaserver.HTTPStatusCode;
+import javaserver.Request;
+import javaserver.Response;
+import javaserver.Routes;
+
+public class DirectoryResponseBuilder implements ResponseBuilder {
+
+	private Response response;
+	private String directory = "public";
+	
+	@Override
+	public Response getResponse(Request request) {
+		this.response = new Response();
+		setResponseData(request);
+		return this.response;
+	}
+
+	@Override
+	public void setResponseData(Request request) {
+		response.setStatusLine(getStatusLine(request));
+		response.setHeader("Content-Type: text/html;");
+		response.setBody(getDirectoryLinksHTML());
+	}
+
+	@Override
+	public String getStatusLine(Request request) {
+		HTTPStatusCode responseCode;
+		if (requestIsSupported(request.getMethod(), request.getURI())) {
+			responseCode = HTTPStatusCode.TWO_HUNDRED;
```

Comment:
You could return the status codes from each conditional branch so that you don't need to assign anything to responseCode.